src/core/thread/mesh_forwarder.cpp

```diff
@@ -755,13 +755,23 @@ Mac::TxFrame *MeshForwarder::HandleFrameRequest(Mac::TxFrames &aTxFrames)
         mMessageNextOffset =
             PrepareDataFrame(*frame, *mSendMessage, mMacAddrs, mAddMeshHeader, mMeshSource, mMeshDest, addFragHeader);
 
-        if ((mSendMessage->GetSubType() == Message::kSubTypeMleChildIdRequest) && mSendMessage->IsLinkSecurityEnabled())
+        if ((mSendMessage->GetSubType() == Message::kSubTypeMleChildIdRequest))
         {
-            LogNote("Child ID Request requires fragmentation, aborting tx");
-            mMessageNextOffset = mSendMessage->GetLength();
-            ExitNow(frame = nullptr);
+            if (mSendMessage->GetTxSuccess())
```

Comment:
The new check is added in `HandleFrameRequest()`.
- `HandleFrameRequest()` is called when the Mac layer is ready to transmit a frame and asks the next layer (`MeshForwarder`) to prepare the frame to be sent.
- At this point, the frame has not yet been sent and the `TxSuccess` status has not yet been updated or set.
- If we want to do such a check, we should probably move it to the `HandleFrameSend()` call path and after the `mSendMessage->SetTxSuccess(false)` call in `MeshForwarder::UpdateSendMessage()`.

Reply:
Thanks @abtink, in `MeshForwarder::UpdateSendMessage()` we are already checking if Child ID request is fragmented and if so requesting for a shorter Child ID Request. Is it ok to move this logic just above that check?

Reply:
Yes. That can work.

Reply:
Thanks @abtink regarding your earlier comments,

do you see a concern for using GetTxSuccess from `MeshForwarder::UpdateSendMessage()` or do you recommend to modify to use some other check? 

For avoiding cross-layer logic (MLE-related decisions in MeshForwarder), wondering if it would be better to add HandleSentMessage in Mle class to handle this logic?`Mle:HandleSentMessage` should then also be invoked from `IndirectSender::HandleSentFrameToChild`right?

Reply:
sure @abtink makes sense and it makes the code cleaner.  tried to make the suggested change, please check if ok. Another thought is we could add a generic HandleTxDone in Mle and move the entire switch statement (where it currently checks kSubTypeMleDiscoverRequest)to Mle using `IsSubTypeMle` check. This would discourage any future code additions to the switch case (like how I had started doing)?